descartes_light/solvers/include/descartes_light/solvers/bgl/impl/event_visitors.hpp

```diff
@@ -32,6 +32,75 @@ struct early_terminator : public boost::base_visitor<early_terminator<FloatType>
   const long last_rung_idx_;
 };
 
+/**
+ * @brief Event visitor that adds all edges to each vertex dynamically as the vertex is discovered during the graph
+ * search
+ */
+template <typename FloatType>
+struct add_all_edges_dynamically : public boost::base_visitor<add_all_edges_dynamically<FloatType>>
+{
+  /** @brief Event filter typedef defining the events for which this visitor can be used */
+  typedef boost::on_discover_vertex event_filter;
```

Comment:
We should make the event filter a template parameter because the more appropriate event for a Dijkstra search is probably `examine_vertex`. For a Dijkstra search, `discover_vertex` is invoked when vertices are *encountered*, meaning this visitor would be adding edges to vertices that were seen by the graph, but not necessarily selected for evaluation, resulting in additional unnecessary work by the solver.